x/concentrated-liquidity/internal/math/math.go

```diff
@@ -81,20 +95,21 @@ func CalcAmount1Delta(liq, sqrtPriceA, sqrtPriceB sdk.Dec, roundUp bool) sdk.Dec
 // else
 // sqrtPriceNext = ((liquidity)) / (((liquidity) / (sqrtPriceCurrent)) + (amountRemaining))
 func GetNextSqrtPriceFromAmount0RoundingUp(sqrtPriceCurrent, liquidity, amountRemaining sdk.Dec) (sqrtPriceNext sdk.Dec) {
-	numerator := liquidity
-	product := amountRemaining.Mul(sqrtPriceCurrent)
+	if amountRemaining.Equal(sdk.ZeroDec()) {
+		return sqrtPriceCurrent
+	}
 
-	if product.Quo(amountRemaining).Equal(sqrtPriceCurrent) {
-		denominator := numerator.Add(product)
-		if denominator.GTE(numerator) {
-			numerator = numerator.Mul(sqrtPriceCurrent)
-			sqrtPriceNext = numerator.QuoRoundUp(denominator)
-			return sqrtPriceNext
+	product := amountRemaining.Mul(sqrtPriceCurrent)
+	// use precise formula if product doesn't overflow
```

Comment:
What is the precise formula?

Why would product overflow?

Reply:
Precise formula is the top one, bottom one is less precise.

I don't know solidity math well enough to answer why their product might overflow

Reply:
Let's not block this PR but create an issue to re-think this entire function and understand the math / whether we need the two kinds of computations